[Benjamin Lui] iP

benluiwj opened the pull request:
# DUKE
Duke frees your mind of having to remember things you need to do. It's,

*  text-based
*  easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is :

1. download it from [here](https://github.com/nus-cs2103-AY2122S1/ip)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practise Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Mainapp.class, args);
    }
}
```

Commit:
Add Gradle support

benluiwj committed:
just testing to see if this works properly

benluiwj committed:
Added level-1

benluiwj committed:
Modified gitignore file

benluiwj committed:
Added javadoc

benluiwj committed:
Corrected javadocs for level1 submission

benluiwj committed:
Implemented Level-2

benluiwj committed:
Fix java and Git conventions

benluiwj committed:
Add level 3

benluiwj committed:
Add level-4

benluiwj committed:
Add TextUITesting

benluiwj committed:
Fix TextUITesting tag

benluiwj committed:
Add level-5

benluiwj committed:
Add level-6

benluiwj committed:
Add A-Enums

benluiwj committed:
Add level-7

benluiwj committed:
Add level-8

benluiwj committed:
Merge branch 'branch-level-7'

benluiwj committed:
Merge branch 'branch-level-8'

benluiwj committed:
Add A-MoreOOP

benluiwj committed:
Add A-Packages

benluiwj committed:
Add A-JUnit

benluiwj committed:
Add A-JavaDoc

benluiwj committed:
Add A-CodingStandard

benluiwj committed:
Add Level-9

benluiwj committed:
Merge branch 'branch-A-JavaDoc'

benluiwj committed:
Merge branch 'branch-A-CodingStandard'

benluiwj committed:
Merge branch 'branch-Level-9'

benluiwj committed:
Merge branch 'add-gradle-support' of https://github.com/benluiwj/Individual-Project

benluiwj committed:
Add A-Gradle

benluiwj committed:
Add A-CheckStyle

benluiwj committed:
Add Level-10

benluiwj committed:
Merge branch 'branch-Level-10'

benluiwj committed:
Add A-Varargs

benluiwj committed:
Merge branch 'branch-A-Varargs'

Domszy commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,29 @@
+import duke.ui.Ui;
+
+
+/**
+ * Contains the executables when the user uses the 'bye' command.
+ */
```
Perhaps you could include the @author tag for your description of the classes. Otherwise i think the description is quite clear!

benluiwj replied:
Yep, ill prolly add it in. Thanks :D

Domszy commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,29 @@
+
+/**
+ * Contains the executables when the user uses the 'bye' command.
+ */
```
Perhaps you could include the @author tag for your description of the classes. Otherwise i think the description is quite clear!

Domszy commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,33 @@
+        storage.save(taskList);
+        try {
+            return ui.doneTask(taskList.getAllTasks().get(taskNum));
+        } catch (IndexOutOfBoundsException ioobe) {
```
huh it's interesting that you named the exception as such! I think it's a good idea to do so though

Domszy commented on src/main/java/duke/command/ToDoCommand.java:
```diff
@@ -0,0 +1,37 @@
+     * Executes the to do command.
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
```
perhaps having a more specific name like "executeTask" could be more useful for other coders to understand; honestly, i'm nitpicking at this point.

Domszy commented on src/main/java/duke/storage/DesAndTime.java:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Processes the task if its an event task.
+     */
+    void event() {
```
Perhaps a more specific name for the event() method as well

Domszy commented on src/main/java/duke/storage/TaskListEncoder.java:
```diff
@@ -0,0 +1,21 @@
+     */
+    public static List<String> encodeTaskList(TaskList toSave) {
+        final List<String> encodedTasks = new ArrayList<String>();
+        toSave.getAllTasks().forEach(task -> encodedTasks.add(task.toString()));
```
The use of the forEach() method to adbstract the code is great; I really like how succint your code is!

Domszy reviewed (commented):
Overall, your code looks quite succint and clear. Great job 🙂

benluiwj committed:
Currently, no author tag for each class.

Author tags notifies the reader who wrote the class.

Added the author tag for all the classes, right after the description
of the class.

Creates a neater description of the class.

benluiwj committed:
Add Author tags

Currently, no author tag for each class.

Author tags notifies the reader who wrote the class.

Added the author tag for all the classes, right after the description
of the class.

Creates a neater description of the class.

benluiwj committed:
Add A-Assertions

Delete method of TaskList class doesn't resolve invalid task number

If there is an invalid task number, there is a possibility that
an exception may be thrown

An assertion was added to ensure that the task number is
smaller than the size of the TaskList

This way, the code breaks if the task number input is invalid.

benluiwj committed:
Merge branch 'master' into branch-A-Assertions

benluiwj committed:
Merge pull request #1 from benluiwj/branch-A-Assertions

Branch a assertions

benluiwj committed:
Add A-CodeQuality

benluiwj committed:
Merge pull request #2 from benluiwj/branch-A-CodeQuality

Add A-CodeQuality

benluiwj committed:
Merge branch 'branch-A-CodeQuality'

benluiwj committed:
Merge branch 'master' of https://github.com/benluiwj/Individual-Project

benluiwj committed:
Add A-Streams

benluiwj committed:
Merge pull request #3 from benluiwj/branch-A-Streams

Add A-Streams

benluiwj committed:
Add A-CI

benluiwj committed:
Revert "Add A-CI"

This reverts commit 05ebe591cc05d3b56ff3581ac9008dad2980506a.

benluiwj committed:
Add A-CI

benluiwj committed:
Merge branch 'branch-A-CI'

benluiwj committed:
Modify gradle.yml

benluiwj committed:
Modify tests to checkstyle standards

benluiwj committed:
Add C-DetectDuplicates

benluiwj committed:
Merge branch 'branch-C-DetectDuplicates'

benluiwj committed:
Modified equals method

benluiwj committed:
Add A-BetterGui

benluiwj committed:
Update credits

benluiwj committed:
Add Ui.png

benluiwj committed:
Add A-UserGuide

benluiwj committed:
Add A-Release

benluiwj committed:
Edit based on code quality feedback
